x-pack/plugins/apm/server/routes/services/get_service_instances/get_service_instances_transaction_statistics.ts

```diff
@@ -48,6 +49,27 @@ type ServiceInstanceTransactionStatistics<T> = T extends true
   ? ServiceInstanceTransactionComparisonStatistics
   : ServiceInstanceTransactionPrimaryStatistics;
 
+export function getOrderInstructions(
+  sortField: InstancesSortField,
+  sortDirection: 'asc' | 'desc'
+): Record<string, 'asc' | 'desc'> {
+  switch (sortField) {
+    case 'errorRate':
+      // To correct order by error rate we need to combine both the 'failures._count' and 'terms._count' fields.
+      // Docs with high failure._count and low terms._count have a high failure rate.
+      return {
+        failures: sortDirection,
+        _count: sortDirection === 'desc' ? 'asc' : 'desc',
+      };
```

Comment:
I don't think will accurately order by error rate. First you are sorting by the number of failures and then you are using the total number of documents as a tie-breaker.

If you have three instances: 
Instance A:  10 failures out of 100 requests (10%)
Instance B: 9 failures out of 10 requests (90%)
Instance C: 8 failures out of 10 requests (80%)

The correct order is: Instance B, Instance C, Instance A
But I think the order with the current implementation will be: Instance A, Instance B, Instance C

The only way to solve this is by calculating error rate in ES using a bucket script like https://discuss.elastic.co/t/using-doc-count-to-calculate-percentage-after-aggregation/243611

Reply:
I'll take a look at this.